### DISCUSSION TITLE
Ensuring copybooks have been downloaded before doing any actions.

### DISCUSSION THREAD

**Discussion by TommyTechh** [IBM/zopeneditor-about#422] [Q&A]
Hi zopenedtor community,

I'm building a vscode extension where i need to ensure that the copybooks found in my .cbl file has been downloaded/resolved before i can call any commands on the .cbl file.

Is there an easy way to "inject" into zopeneditor and check if it has resolved all the copybooks, or do i need to manually check the downloads in the temp folder?

Is this a question that should be asked at zowe or vscode instead?

Thanks in advance

**Comment by phaumer** [comment]
My apologies for the slow reply. We do not expose APIs for accessing our language server, but you might be able to access the problems view via VS Code APIS, e.g. via the diagnostics collection (have not tried that though). If you get access to the diagnostics you can search for the "Unable to find copybook" messages. The downside is that the program has to be open in the editor.
